include/posix/pthread.h

```diff
@@ -383,4 +393,151 @@ int pthread_barrierattr_getpshared(const pthread_barrierattr_t *, int *);
 int pthread_barrierattr_setpshared(pthread_barrierattr_t *, int);
 */
 
+/* Base Pthread related APIs */
+
+/**
+ * @brief get pid of current execueting thread.
+ *
+ * See IEEE 1003.1
+ */
+static inline pthread_t pthread_self(void)
+{
+	return (pthread_t)k_current_get();
+}
+
+/**
+ * @brief POSIX pthread attributes destroy
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_destroy(pthread_attr_t *attr)
+{
+	return 0;
+}
+
+/**
+ * @brief POSIX get pthread scheduling parameter
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getschedparam(const pthread_attr_t *attr,
+					     struct sched_param *schedparam)
+{
+	schedparam->priority = attr->priority;
+	return 0;
+}
+
+/**
+ * @brief POSIX get pthread stack inforamtion from thread attribute
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getstack(const pthread_attr_t *attr,
+					void **stackaddr, size_t *stacksize)
+{
+	*stackaddr = attr->stack;
+	*stacksize = attr->stacksize;
+	return 0;
+}
+
+/**
+ * @brief POSIX get stack size from thread attribute
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getstacksize(const pthread_attr_t *attr,
+					    size_t *stacksize)
+{
+	*stacksize = attr->stacksize;
+	return 0;
+}
+/**
+ * @brief are Both pthreads same
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_equal(pthread_t pt1, pthread_t pt2)
+{
+	return (pt1 == pt2);
+}
+
+/**
+ * @brief get pid of current execueting thread.
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_setschedpolicy(pthread_attr_t *attr, int policy)
+{
+	if (policy != SCHED_RR && policy != SCHED_FIFO) {
+		errno =  EINVAL;
+		return -1;
+	}
+
+	attr->schedpolicy = policy;
+	return 0;
+}
+
+/**
+ * @brief get pid of current execueting thread.
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getschedpolicy(const pthread_attr_t *attr,
+					      int *policy)
+{
+	*policy = attr->schedpolicy;
+	return 0;
+}
+
+/**
+ * @brief get pid of current execueting thread.
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_setdetachstate(pthread_attr_t *attr,
+					       int detachstate)
+{
+	if ((attr == NULL) || (detachstate != PTHREAD_CREATE_DETACHED &&
```

Comment:
The handling of attr being a null pointer is irregular here.  Either check everywhere (could be an assert), or don't check at all: the standard says that the behavior is undefined if attr does not refer to an initialized thread attributes object.

You might want to study the [nonnull attribute](https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes) in GCC as an alternative.

Reply:
Addressed